Nasal/FMGC/FMGC.nas

```diff
@@ -603,88 +605,144 @@ var radios = maketimer(1, func() {
-			systems.PNEU.pressMode.setValue("DE");
-		}
-	} else {
-		if (FMGCInternal.phase == 3 and (flightPlanController.arrivalDist <= 200 or altSel < (FMGCInternal.crzFl * 100))) { # todo - not sure about crzFl condition, investigate what happens!
-			FMGCInternal.phase = 4;
-			systems.PNEU.pressMode.setValue("DE");
+	else if (phase == 2) {
+
+		if ((mode == "ALT CRZ" or mode == "ALT CRZ*")) {
+			newphase = 3;
+			systems.PNEU.pressMode.setValue("CR");
 		}
+	
 	}
+
+	else if (phase == 3) {
+
+		if (FMGCInternal.crzFl >= 200) {
+			if ((flightPlanController.arrivalDist <= 200 or altSel < 20000)) {
+				newphase = 4;
+				systems.PNEU.pressMode.setValue("DE");
+			}
+		} else {
+			if ((flightPlanController.arrivalDist <= 200 and altSel < (FMGCInternal.crzFl * 100))) { # todo - not sure about crzFl condition, investigate what happens!
+				newphase = 4;
+				systems.PNEU.pressMode.setValue("DE");
+			}
+		}
 	
-	if (FMGCInternal.phase == 4) {
+	}
+
+	else if (phase == 4) {
+
 		if (getprop("/FMGC/internal/decel")) {
-			FMGCInternal.phase = 5;
+			newphase = 5;
 		}
 		else if (altSel == (FMGCInternal.crzFl * 100)) {  # back to CRZ state
-			FMGCInternal.phase = 3;
+			newphase = 3;
 			systems.PNEU.pressMode.setValue("CR");
 		}
+
 	} 
 
-	if (flightPlanController.num[2].getValue() > 0 and getprop("/FMGC/flightplan[2]/active") == 1 and flightPlanController.arrivalDist <= 15 and (modelat == "NAV" or modelat == "LOC" or modelat == "LOC*") and pts.Position.gearAglFt.getValue() < 9500) { #todo decel pseudo waypoint
-		setprop("/FMGC/internal/decel", 1);
-	} else if (getprop("/FMGC/internal/decel") == 1 and (FMGCInternal.phase == 0 or FMGCInternal.phase == 6)) {
-		setprop("/FMGC/internal/decel", 0);
-	}
+	else if (phase == 5) {
+	
+		if (state1 == "TOGA" and state2 == "TOGA") {
+			newphase = 6;
+			systems.PNEU.pressMode.setValue("TO");
+			Input.toga.setValue(1);
+		}
 	
-	if ((FMGCInternal.phase == 5) and state1 == "TOGA" and state2 == "TOGA") {
-		FMGCInternal.phase = 6;
-		systems.PNEU.pressMode.setValue("TO");
-		Input.toga.setValue(1);
 	}
 	
-	if (FMGCInternal.phase == 6 and alt >= accel_agl_ft) { # todo when insert altn or new dest
-		FMGCInternal.phase = 2;
+	else if (phase == 6) {
+
+		if (alt >= accel_agl_ft) { # todo when insert altn or new dest
+			newphase = 2;
+		}
+
 	}
+
+
+	if (flightPlanController.num[2].getValue() > 0 and getprop("/FMGC/flightplan[2]/active") == 1 and 
+	   flightPlanController.arrivalDist <= 15 and (modelat == "NAV" or modelat == "LOC" or modelat == "LOC*") and pts.Position.gearAglFt.getValue() < 9500) { #todo decel pseudo waypoint
+		setprop("/FMGC/internal/decel", 1);
+	} else if (getprop("/FMGC/internal/decel") == 1 and (phase == 0 or phase == 6)) {
+		setprop("/FMGC/internal/decel", 0);
+	}
+	
 	
-	tempOverspeed = systems.ADIRS.overspeedVFE.getValue();
-	if (tempOverspeed != 1024) {
-		FMGCInternal.maxspeed = tempOverspeed - 4;
+	if (getprop("/systems/navigation/adr/computation/overspeed-vfe-spd") != 1024) {
```

Comment:
please don't revert my changes :)  should use systems.ADIRS.overspeedVFE.getValue();

Reply:
Sorry, I dunno I lost your changes. There is some other changes missing?